rascal-vscode-extension/src/auto-jvm/downloaders.ts

```diff
@@ -181,7 +182,33 @@
     return `jdk-${rel.openjdk_version}`;
 }
 
-function mapTemuringCorrettoArch(): TemurinArchitectures {
+export async function identifyLatestCorrettoRelease(version: number, arch: TemurinArchitectures, platform: CorrettoPlatforms): Promise<string> {
+    const url = `https://corretto.aws/downloads/latest/amazon-corretto-${version}-${arch}-${platform}-jdk.${platform === "windows" ? "zip" : "tar.gz"}`;
+    const response = await fetch(url, {method: 'HEAD'});
+    const fileUrl = response.url;
+    const versionPattern = /resources\/((\d*\.?)+)\//;
+    const match = fileUrl.match(versionPattern);
+    if(!match){
+        throw new Error(`unexpected response url ${response}`);
+    }
+    return match[1];
+}
+
+
+// Does not identify +x versions
+export async function identifyLatestMicrofotJDKRelease(version: number, arch: MSArchitectures, platform: MSPlatforms): Promise<string> {
+    const url = `https://aka.ms/download-jdk/microsoft-jdk-${version}-${platform}-${arch}.${platform === "windows" ? "zip" : "tar.gz"}`;
+    const response = await fetch(url, {method: 'HEAD'});
+    const fileUrl = response.url;
+    const versionPattern = /jdk-((\d*\.?)+)-/;
```

Comment:
I think the regular expression has a bit too many optional parts. since \.? can match 0 or 1 . and \d* can match zero or more chars. maybe we can write it more like: /jdk-[0-9\.]+-/?